packages/esm-ward-app/src/ward-patient-card/ward-patient-card.component.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useMemo } from 'react';
+import { type Encounter, type Bed } from '../types';
+import { WardPatientCardElement } from './ward-patient-card-element.component';
+import { useCurrentWardCardConfig } from '../hooks/useCurrentWardCardConfig';
+import styles from './ward-patient-card.scss';
+import { ExtensionSlot, getPatientName, launchWorkspace, type Patient, type Visit } from '@openmrs/esm-framework';
+import WardPatientName from './row-elements/ward-patient-name';
+import WardPatientBedNumber from './row-elements/ward-patient-bed-number';
+import classNames from 'classnames';
+import { type WardPatientWorkspaceProps } from '../ward-patient-workspace/types';
+
+export interface WardPatientCardProps {
+  patient: Patient;
+  visit: Visit;
+  bed?: Bed;
+  admitted: boolean;
+  encounterAssigningToCurrentInpatientLocation: Encounter;
+  firstAdmissionOrTransferEncounter: Encounter;
+}
+
+export interface WardPatientCardExtensionProps extends WardPatientCardProps {
+  patientUuid: string;
+}
+
+const WardPatientCard: React.FC<WardPatientCardProps> = ({
+  patient,
+  visit,
+  bed,
+  admitted,
+  firstAdmissionOrTransferEncounter,
+  encounterAssigningToCurrentInpatientLocation,
+}) => {
+  const { id, headerRowElements, footerRowElements } = useCurrentWardCardConfig();
+
+  const headerExtensionSlotName =
+    id == 'default' ? 'ward-patient-card-header-slot' : `ward-patient-card-header-${id}-slot`;
+  const rowsExtensionSlotName = id == 'default' ? 'ward-patient-card-slot' : `ward-patient-card-${id}-slot`;
+  const footerExtensionSlotName =
+    id == 'default' ? 'ward-patient-card-footer-slot' : `ward-patient-card-footer-${id}-slot`;
+
+  const extensionSlotState = useMemo(() => {
+    return {
+      patient,
+      visit,
+      bed,
+    };
+  }, [patient, visit, bed]);
+
+  return (
+    <div className={styles.wardPatientCard}>
+      <div className={classNames(styles.wardPatientCardRow, styles.wardPatientCardHeader)}>
+        {bed ? <WardPatientBedNumber bed={bed} /> : null}
+        <WardPatientName patient={patient} />
+        {headerRowElements.map((elementId, i) => (
+          <WardPatientCardElement
+            key={`ward-card-${patient.uuid}-header-${i}`}
+            elementId={elementId}
+            patient={patient}
+            visit={visit}
+            firstAdmissionOrTransferEncounter={firstAdmissionOrTransferEncounter}
+            encounterAssigningToCurrentInpatientLocation={encounterAssigningToCurrentInpatientLocation}
+          />
+        ))}
+        <ExtensionSlot name={headerExtensionSlotName} state={extensionSlotState} />
```

Comment:
I was hoping there is an `ExtensionElement` that can be placed with other elements in arbitrary instead of at the end, but we can do that in a different PR.

Reply:
I don't understand, could you elaborate on this?

Reply:
Sorry, typo. should be "in arbitrary order". So we can have, for example, `bed-number`, `patient-name`, `extension-element`, `patient-identifier` (instead of having the `extension-element` at the end)

Reply:
If that becomes a real need, the thing to do would probably be to turn the elements that come after the patient name into extensions and have an extension slot come after name. I would suggest not treating bed number and patient name as arbitrary "bento box" elements, but rather as things that are managed by the card itself. So in the main card you'd have

bed number    |   patient name    |  header extension slot

And then in the admission request card you'd have

patient name    |  header extension slot

And then all of those elements, obs display etc, would be configurable directly using extension config (so the card wouldn't have anything to do with their configuration), and you'd just use the config system to add/remove/rearrange them.

But that would come with some performance cost (which would be significantly lightened by [O3-1418](https://openmrs.atlassian.net/browse/O3-1418)). So I'd recommend not doing that until it's clear that there is a use case.